src/Components/VideoSearchApp.js

```diff
@@ -0,0 +1,56 @@
+import React, { useState } from "react";
+import VideoSearchList from "./VideoSearchList";
+import Player from "./Player";
+import VideoSearch from "./VideoSearch";
+
+import { Container, Row, Col } from "react-bootstrap/";
+
+const VideoSearchApp = () => {
+  const [movies, setMovies] = useState([]);
+  const [movieId, setMovieId] = useState("");
+
+  const searchResultHandler = async (searchText) => {
+    const key = "AIzaSyBsGroGmofzjHBtIG_y4ugaJw8bqSp9eaY";
+    const response = await fetch(
+      "https://youtube.googleapis.com/youtube/v3/search?part=snippet&maxResults=25&q=" +
+        searchText +
+        "&key=" +
+        key
+    );
+
+    let responseData = await response.json();
+    let movieList = [];
+
+    for (const key in responseData.items) {
+      movieList.push({
+        id: key,
+        videoId: responseData.items[key].id.videoId,
+        title: responseData.items[key].snippet.title,
+        thumbnail: responseData.items[key].snippet.thumbnails,
+      });
+    }

```

Comment:
`for / while` loops aren't used anymore. Make use of `map` or `forEach` depending upon the need.

```suggestion
    movies = responseData.items.map(movie => ({
        id: key,
        videoId: responseData.items[key].id.videoId,
        title: responseData.items[key].snippet.title,
        thumbnail: responseData.items[key].snippet.thumbnails,
      }));
```

Also, it's better to use `movies` as opposed to `movieList`. See how I used `movie` inside the map because it represented one item.